pkg/validator/app.go

```diff
@@ -306,21 +306,42 @@ func validateRoles(
 				},
 			)
 		}
-		if role.PersistDirs == nil {
-			if globalPersistDirs != nil {
-				role.PersistDirs = globalPersistDirs
-				patches = append(
-					patches,
-					appPatchSpec{
-						Op:   "add",
-						Path: "/spec/roles/" + strconv.Itoa(index) + "/persistDirs",
-						Value: appPatchValue{
-							stringSliceValue: globalPersistDirs,
-						},
-					},
-				)
+
+		// If the role spec doesn't contain individual persistent dirs
+		// copy them from global application spec
+		if role.PersistDirs == nil && globalPersistDirs != nil {
+			role.PersistDirs = globalPersistDirs
+		}
+
+		// EZML-865
+		// For each persistent dir defined for the role add
+		// corresponding old_<directory_name> persistent dir for backup data
+		// for upgrade
```

Comment:
We can talk about this in the meeting, curious to hear more about the thoughts behind this.

Reply:
As discussed in meeting, let's omit this aspect for now.

Reply:
Discarded